all/plugin-dialects/dialectsplugin/__main__.py

```diff
@@ -0,0 +1,252 @@
+                            ),
+                            size_global=int(
+                                np.sum(dialect_feature_files[dialect_index, :])
+                            ),
+                            highlight=(
+                                bool(
+                                    dialect_feature_files[
+                                        dialect_index, highlight_file_index
+                                    ]
+                                )
+                                if highlight_file_index is not None
+                                else False
+                            ),
+                            filenames_outside_target=[
+                                (
+                                    filenames[file_index],
+                                    bool(file_index == highlight_file_index),
+                                )
+                                for file_index in np.nonzero(
+                                    np.delete(
+                                        dialect_feature_files[dialect_index, :],
+                                        target_files,
+                                    )
+                                )[0]
+                            ],
+                        )
+                        for dialect_index, (hero_feature_index, inverted) in enumerate(
+                            zip(partition.hero_features, partition.inverted)
+                        )
+                    ],
+                    partition_quality={
+                        metric_name: float(
+                            metric_func(
+                                partition_file_indices,
+                                target_feature_files,
+                            )
+                        )
+                        for metric_name, metric_func in quality_metrics.items()
+                    },
+                    slop_files=[
+                        SlopFile(
+                            filename=filenames[
+                                file_index := target_files[target_file_index]
+                            ],
+                            in_dialects=(
+                                dialect_feature_files[:, file_index]
+                                .nonzero()[0]
+                                .tolist()
+                            ),
+                            highlight=bool(file_index == highlight_file_index),
+                        )
+                        for target_file_index in (
+                            target_file_dialect_membership_counts != 1
+                        ).nonzero()[0]
+                    ],
+                )
+            )
+    else:
+        partitions = []
+        debug_lines = []
+        target_size = 0
+
+    with open(output_html, "w") as f, open("./assets/vue.js") as f_vue, open(
+        "./assets/main.js"
+    ) as f_main, open("./assets/style.css") as f_style, open(
```

Comment:
Is this the way `black` formats it? The two `as` on a single line -- with one's body on another line -- is pretty abrasive on the eyes

Reply:
I think this is a relic from when the FAW was on python 3.8! Parenthesized context managers were only added in a more recent version. I originally had this written with backslashes, and `black` mangled it. Now that parens are supported, I'll use them.